src/sql/src/plan/statement.rs

```diff
@@ -578,14 +584,37 @@ fn handle_create_sink(
     let as_of = as_of.map(|e| query::eval_as_of(scx, e)).transpose()?;
     let connector_builder = match connector {
         Connector::File { .. } => unsupported!("file sinks"),
-        Connector::Kafka { broker, topic } => kafka_sink_builder(
-            format,
-            with_options,
+        Connector::Kafka {
             broker,
             topic,
-            from.desc()?.clone(),
-            suffix,
-        )?,
+            keys,
+        } => {
+            let desc = from.desc()?;
+            let key_indices = if let Some(keys) = keys {
+                let idces = keys
+                    .iter()
+                    .map(|k| {
+                        desc.iter_names()
+                            .position(|name| {
```

Comment:
so according to comments in the `RelationDesc` implementation, multiple columns can have the same name, in which case this code takes the leftmost index. Not sure if that was an intentional choice but if so we should definitely put it in the user facing docs I think

Reply:
That's a good point, but I'm not sure what else we can do since I'm not really up to date on how names/identifiers/scoping/etc. is supposed to work in Materialize. Should we allow people to disambiguate between the various ambiguous columns somehow, just print out a warning if there are ambiguous columns, or what?

ruchir and @benesch , any thoughts?

Reply:
Talked to Nikhil in person and we agreed to reject keys whose column names are ambiguous.